src/coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -25,6 +25,60 @@
 
 namespace coreneuron {
 
+void check_version(const std::string& raw,
```

Comment:
I am not terribly happy with the signature of this. At the risk of overengineering, I would say a class with dedicated methods (including string formatting) to parse version info of `report.conf` would be much more maintainable.

Reply:
this is really over-engineering but ok. You want a class that:
- takes a string. Understands the version
- compare to another class (hardcoded) and complains if versions differ. 

Is it correct?

Ok for `fmt::format`